src/plugins/intel_gna/gna_plugin.cpp

```diff
@@ -490,42 +456,82 @@ void GNAPlugin::UpdateInputsAndOutputsInfoFromNetwork(InferenceEngine::CNNNetwor
-
-        size_t outputIdx = 0;
         for (const auto output : outputs) {
-            outputsDesc[outputIdx].precision = output.second->getPrecision().getPrecVal();
-            ++outputIdx;
+            outputs_[output.first].Update(output.second);
         }
     }
 }
 
-bool GNAPlugin::TryToInitOutput(int portId, InferenceEngine::CNNLayerPtr layer) {
-    auto initOutput = [this, portId, layer]
+void GNAPlugin::UpdateInputs(const std::vector<std::shared_ptr<const ov::Node>>& params) {
+    OV_ITT_SCOPED_TASK(itt::domains::GNA_LT, "UpdateInputs");
+    for (const auto& param : params) {
+        const std::string ie_name = param->get_friendly_name();
+        (*inputs_ptr_)[ie_name].name = param->get_friendly_name();
+        (*inputs_ptr_)[ie_name].tensor_names = param->get_output_tensor(0).get_names();
+    }
+}
+
+void GNAPlugin::UpdateOutputs(const std::vector<std::shared_ptr<const ov::Node>>& results) {
+    OV_ITT_SCOPED_TASK(itt::domains::GNA_LT, "UpdateOutputs");
+    for (const auto& result : results) {
+        const std::string ie_name = ngraph::op::util::create_ie_output_name(result->input_value(0));
+        outputs_[ie_name].name = ie_name;
+        outputs_[ie_name].tensor_names = result->get_output_tensor(0).get_names();
+    }
+}
+
+void GNAPlugin::UpdateInputsAndOutputsInfoFromModel(const std::shared_ptr<ov::Model> &model) {
+    OV_ITT_SCOPED_TASK(itt::domains::GNA_LT, "UpdateInputsAndOutputsInfoFromFModel");
+
+    // update inputs
+    {
+        std::vector<std::shared_ptr<const ov::Node>> node_vector;
+        for (auto& param : model->get_parameters()) {
+            node_vector.emplace_back(param);
+        }
+        UpdateInputs(node_vector);
+    }
+
+    // update outputs
+    {
+        std::vector<std::shared_ptr<const ov::Node>> node_vector;
+        for (auto& result : model->get_results()) {
+            node_vector.emplace_back(result);
+        }
+        UpdateOutputs(node_vector);
+    }
+}
+
+bool GNAPlugin::TryToInitOutput(const std::string &portName, InferenceEngine::CNNLayerPtr layer) {
+    auto initOutput = [this, portName, layer]
             (intel_dnn_orientation_t orientation, size_t numBytesPerElem, size_t numElem, void* outputPtr) {
-        auto & desc = outputsDesc[portId];
         auto quantized = InferenceEngine::getInjectedData<QuantizedLayerParams>(layer);
 
-        desc.ptrs.resize(gnaFlags->gna_lib_async_threads_num);
-        desc.orientation = orientation;
-        desc.num_bytes_per_element = numBytesPerElem;
-        desc.scale_factor = quantized != nullptr ? quantized->_dst_quant.GetScale() : 1.0f;
-        desc.num_elements = numElem;
+        outputs_.at(portName).ptrs.resize(gnaFlags->gna_lib_async_threads_num);
+        outputs_.at(portName).orientation = orientation;
+        outputs_.at(portName).num_bytes_per_element = numBytesPerElem;
+        outputs_.at(portName).scale_factor = quantized != nullptr ? quantized->_dst_quant.GetScale() : 1.0f;
```

Comment:
GNAPluginNS::kScaleFactorDefault can be used here instead of 1.0

Reply:
Yes, thanks!

Reply:
fixed